app/src/main/java/ingsw/group1/findmyphone/fragment/LogFragment.java

```diff
@@ -59,12 +69,109 @@ public View onCreateView(
             @NonNull LayoutInflater inflater,
             ViewGroup container,
             Bundle savedInstanceState) {
+
         View root = inflater.inflate(R.layout.log_fragment, container, false);
-        RecyclerView logRecycler = root.findViewById(R.id.log_recycler);
+
+        // Recycler setup --------------------------------------------------------------------------
+        logRecycler = root.findViewById(R.id.log_recycler);
         logRecycler.setLayoutManager(new LinearLayoutManager(getContext()));
         LogRecyclerAdapter logAdapter = new LogRecyclerAdapter(container.getContext(), logManager);
         logRecycler.setAdapter(logAdapter);
         logManager.setListener(logAdapter);
+
+        // Sort button setup -----------------------------------------------------------------------
+        sortButton = root.findViewById(R.id.sort_button);
+        registerForContextMenu(sortButton);
+        sortButton.setOnClickListener(
+                new View.OnClickListener() {
+                    @Override
+                    public void onClick(View view) {
+                        showMenu(view);
+                    }
+                }
+        );
+
+        // Search view setup -----------------------------------------------------------------------
+        searchView = root.findViewById(R.id.searchView);
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String s) {
+                logManager.filter(s);
+                return true;
+            }
+
+            @Override
+            public boolean onQueryTextChange(String s) {
+                logManager.filter(s);
+                return true;
+            }
+        });
+
         return root;
     }
+
+    /**
+     * When the fragment is out of focus, the manager does not need to update the recycler.
+     */
+    @Override
+    public void onPause() {
+        logManager.removeListener();
+        super.onPause();
+    }
+
+    /**
+     * When the fragment is resumed we ensure that, if the adapter was not deleted, it starts
+     * listening to the manager again.
+     */
+    @Override
+    public void onResume() {
+        if (logRecycler.getAdapter() != null)
+            logManager.setListener(logRecycler.getAdapter());
+        super.onResume();
+    }
+
+    /**
+     * Method called to inflate a popup menu and attach it to a View.
+     * Needs to be called by passing {@link LogFragment#sortButton} and for the fragment's
+     * {@link Context} to not be {@code null} in order to proceed.
+     *
+     * @param anchorView The view to which the menu will be attached.
+     */
+    public void showMenu(View anchorView) {
+        if (getContext() == null || anchorView != sortButton) return;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use equals() to compare object references.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5028692)